crates/api/src/local_user/reset_password.rs

```diff
@@ -6,23 +6,26 @@ use lemmy_api_common::{
   SuccessResponse,
 };
 use lemmy_db_views::structs::{LocalUserView, SiteView};
-use lemmy_utils::error::{LemmyErrorExt, LemmyErrorType, LemmyResult};
+use lemmy_utils::error::LemmyResult;
+use tracing::error;
 
 #[tracing::instrument(skip(context))]
 pub async fn reset_password(
   data: Json<PasswordReset>,
   context: Data<LemmyContext>,
 ) -> LemmyResult<Json<SuccessResponse>> {
-  // Fetch that email
   let email = data.email.to_lowercase();
-  let local_user_view = LocalUserView::find_by_email(&mut context.pool(), &email)
-    .await
-    .with_lemmy_type(LemmyErrorType::IncorrectLogin)?;
-
-  let site_view = SiteView::read_local(&mut context.pool()).await?;
-  check_email_verified(&local_user_view, &site_view)?;
-
-  // Email the pure token to the user.
-  send_password_reset_email(&local_user_view, &mut context.pool(), context.settings()).await?;
+  // Fetch that email
+  if let Ok(local_user_view) = LocalUserView::find_by_email(&mut context.pool(), &email).await {
+    if let Ok(site_view) = SiteView::read_local(&mut context.pool()).await {
+      if check_email_verified(&local_user_view, &site_view).is_ok() {
+        if let Err(e) =
+          send_password_reset_email(&local_user_view, &mut context.pool(), context.settings()).await
+        {
+          error!("Failed to send password reset email: {}", e);
+        }
+      }
+    }
+  }
```

Comment:
Instead of changing this to nested if statements, move everything to a separate `try_reset_password` function, and ignore the errors returned by that